### PR TITLE
Issue 18 UI

### PR DESCRIPTION
**Pull request by pulkitsinghall**
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have added tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [x] I have Extended README or added documentation/comments where applicable
- [x] Added copyright headers for new files from CONTRIBUTING.md
- [x] My code follows the code style of this project.

- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have updated the [ChangeLog](CHANGELOG.md) following [Semantic Versioning](https://semver.org)

### CODE REVIEW

File: Daksha_Recorder_with_UI/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/.gitignore:6]: change to daksha_recorder_UI

File: Daksha_Recorder_with_UI/public/EventPage.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+//             "type": "resume"
+//         }
+//         chrome.tabs.sendMessage(tab.id, obj, () => { return true; });
+//         chrome.contextMenus.removeAll();
+//         chrome.contextMenus.create(pause);
+//         chrome.contextMenus.create(stopRecording);
+//         chrome.contextMenus.create(download);
+//         chrome.contextMenus.create(copy_to_clipboard);
+//         chrome.contextMenus.create(hard_wait);
+//         chrome.contextMenus.create(customSecondsWait);
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/public/EventPage.js:164]: remove commented code

File: Daksha_Recorder_with_UI/public/RecordingTab.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+window.addEventListener("mousedown" , (e)=>{
+    console.log('i am called.........')
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/public/RecordingTab.js:2]: remove console logs

File: Daksha_Recorder_with_UI/src/Components/badgeForRecording.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/*global chrome*/
+function badgeForRecording() {
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/Components/badgeForRecording.js:2]: change this to setBadgeForRecording

File: Daksha_Recorder_with_UI/src/ComponentCss/style.css [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    display: flex;
+    justify-content: space-evenly;
+    flex-direction: column;
+    height: 100px;
+    width: 250px;
+    font-size: medium;
+    font-family:Aktiv Grotesk,-apple-system,BlinkMacSystemFont,San Francisco,Segoe UI,Roboto,Helvetica Neue,sans-serif;
+}
+
+.secondDiv{
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/ComponentCss/style.css:11]: meaningful naming required

File: Daksha_Recorder_with_UI/src/ComponentCss/popup2.css [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    height: 150px;
+    width: 215px;
+    font-size: medium;
+    user-select: none;
+}
+
+.popup2-input-container{
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/ComponentCss/popup2.css:11]: popup2 name change

**Review by hriddhighosh4 — APPROVED** [review]

File: Daksha_Recorder_with_UI/public/ContentScript.js [hunk 7]
```diff
@@ -0,0 +1 @@
...if("pause"===t.type){var r={updatePauseValue:!0};chrome.storage.sync.set(r,(()=>{})),s=!0,h()}else"start"===t.type?(chrome.storage.sync.remove(c,(()=>{})),o="",a="",l="",e=null,r={updatePauseValue:!1},chrome.storage.sync.set(r,(()=>{})),s=!1,g([(u=t.msg,{open_url:{url:`${u}`}})]),d()):"resume"===t.type?(r={updatePauseValue:!1},chrome.storage.sync.set(r,(()=>{})),s=!1,d()):"customSecondsWait"===t.type?(g([y(t.sec)]),alert(`${t.sec} seconds hard wait added!`)):"tenSecondsWait"===t.type?(g([y(10)]),alert("10 secs hard wait added!")):"stop"===t.type?!1===s&&(r={updatePauseValue:!0},chrome.storage.sync.set(r,(()=>{})),s=!0,h()):"viewYaml"===t.type?chrome.storage.sync.get(c,(function(e){var t=`<iframe width="100%" height="100%" srcdoc="<pre>${C(e.generatedDakshaYaml)}</pre>"></iframe>`,n=window.open("","","_blank");n.document.open(),n.document.write(t)})):"undoLastStep"===t.type&&(chrome.storage.sync.get(c,(function(e){var t=e.generatedDakshaYaml;if(t.length>1){t.pop();var n={};n.generatedDakshaYaml=t,chrome.storage.sync.set(n,(()=>{}))}})),alert("Last step has been removed!"));var u;return i({msg:"Request Processed"}),Promise.resolve("Dummy response to keep the console quiet")}))}()}();
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/public/ContentScript.js:1]: Please use proper formatting

File: Daksha_Recorder_with_UI/public/EventPage.js [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+    })
+}
+let removeBadgeForRecording = () => {
+    chrome.action.setBadgeBackgroundColor({ color: '#F00' }, () => {
+        chrome.action.setBadgeText({ text: '' });
+    })
+}
+//Installing Context Menus at the beginning of chrome Extension Installed.
+chrome.runtime.onInstalled.addListener(() => {
+    //chrome.contextMenus.create(start);
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/public/EventPage.js:80]: remove commented out code

File: Daksha_Recorder_with_UI/src/App.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
+/*global chrome*/
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import DakshaRecorderStartingPage from './Components/dakshaRecorderStartingPage';
+import DakshaRecorderMainPage from './Components/dakshaRecorderMainPage';
+import DakshaRecorderCustomHardwaitPage from './Components/dakshaRecorderCustomHardwaitPage';
+import DakshaRecorderEndPage from './Components/dakshaRecorderEndPage';
+
+let pageNumber;
+chrome.storage.sync.get("popupPageNumber", function (result) {
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/App.js:10]: remove string and use final variables

File: Daksha_Recorder_with_UI/src/App.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    )
+  else if (state == 2)
+    return (
+      <DakshaRecorderMainPage setState={setState} image={image} changeImage={changeImage}/>
+    )
+  else if (state == 3)
+    return (
+      <DakshaRecorderCustomHardwaitPage setState={setState} />
+    )
+  else if (state == 4)
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/App.js:56]: Can states be changed to string.I assume you are using them for pageNumber 1,2,3,4

File: Daksha_Recorder_with_UI/src/Components/dakshaRecorderCustomHardwaitPage.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import React, { useState } from 'react';
+import '../ComponentCss/dakshaRecorderCustomHardwaitPage.css';
+function DakshaRecorderCustomHardwaitPage(props) {
+    const [val, setval] = useState();
+    const func = (value) => {
+        if (value >= 0) {
+            setval(value);
+        }
+    }
+    return (
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/Components/dakshaRecorderCustomHardwaitPage.js:11]: Why are we directly returning html ?

File: Daksha_Recorder_with_UI/src/Components/dakshaRecorderEndPage.js [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+import React from "react";
+import '../ComponentCss/dakshaRecorderEndPage.css';
+import badgeForRecording from "./badgeForRecording";
+
+function DakshaRecorderEndPage(props) {
+    return (
+        <>
+            <div className="end-page-container">
+                <div>
+                    Daksha Recorder
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/Components/dakshaRecorderEndPage.js:11]: why are we returning html directly

File: Daksha_Recorder_with_UI/src/ContentScript.js [hunk 13]
```diff
@@ -0,0 +1,491 @@
...
+        var array = result[dakshaYamlStorageKey] ? result[dakshaYamlStorageKey] : [];
+        array.push.apply(array, dakshaYamlObjects);
+        var dakshaYamlObject = {};
+        dakshaYamlObject[dakshaYamlStorageKey] = array;
+        chrome.storage.sync.set(dakshaYamlObject, () => {
+        });
+        // getYamlFileData(array) ;
+    });
+}
+// Defined the dakshaYamlObjectss which will be added into the updateDakshaYamlFile function!!
```
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/src/ContentScript.js:192]: remove comments

File: Daksha_Recorder_with_UI/public/ContentScript.js (hunk 7 above)
Comment (hriddhighosh4) [inline Daksha_Recorder_with_UI/public/ContentScript.js:1]: Change project structure to recorder/extensions/chrome/public

File: Daksha_Recorder_with_UI/public/RecordingTab.js (hunk 3 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/public/RecordingTab.js:2]: done

File: Daksha_Recorder_with_UI/src/Components/badgeForRecording.js (hunk 4 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/src/Components/badgeForRecording.js:2]: done

File: Daksha_Recorder_with_UI/src/ComponentCss/style.css (hunk 5 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/src/ComponentCss/style.css]: done

File: Daksha_Recorder_with_UI/src/ComponentCss/popup2.css (hunk 6 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/src/ComponentCss/popup2.css]: done

File: Daksha_Recorder_with_UI/public/ContentScript.js (hunk 7 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/public/ContentScript.js:1]: done

File: Daksha_Recorder_with_UI/public/EventPage.js [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+    "contexts": ["all"]
+}
+let stopRecording = {
+    "id": "stop",
+    "title": "Stop",
+    "contexts": ["all"]
+}
+
+let badgeForRecording = () => {
+    chrome.action.setBadgeBackgroundColor({ color: '#F00' }, () => {
```
[the remark this answers is not in the document]
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/public/EventPage.js:69]: Changed name to setBadgeForRecording

File: Daksha_Recorder_with_UI/public/EventPage.js (hunk 8 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/public/EventPage.js:80]: done

File: recorder/extensions/chrome/public/RecordingTab.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+window.addEventListener("mousedown" , (e)=>{
+    console.log('i am called.........')
```
Comment (hriddhighosh4) [inline recorder/extensions/chrome/public/RecordingTab.js:2]: remove this console log

File: recorder/extensions/chrome/src/Components/dakshaRecorderEndPage.js [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+                                "type": "copy_to_clipboard"
+                            }
+                            chrome.tabs.sendMessage(tabs[0].id, obj, () => { return true; });
+                        })
+                    }}>
+                        COPY TO CLIPBOARD
+                    </div>
+                    <div className="end-page-each-option-div" id="end-page-recording-button" onClick={() => {
+                        props.changeImage(1);
+                        props.setState(2);
```
Comment (hriddhighosh4) [inline recorder/extensions/chrome/src/Components/dakshaRecorderEndPage.js:38]: change this states to int  variable

File: Daksha_Recorder_with_UI/src/App.js (hunk 10 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/src/App.js]: done

File: recorder/extensions/chrome/public/EventPage.js [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+//             "type": "pause"
+//         }
+//         chrome.tabs.sendMessage(tab.id, obj, () => { return true; });
+//         chrome.contextMenus.removeAll();
+//         chrome.contextMenus.create(stopRecording);
+//         chrome.contextMenus.create(resume);
+//         chrome.contextMenus.create(copy_to_clipboard);
+//         chrome.contextMenus.create(download);
+//         removeBadgeForRecording();
+//     }
```
[the remark this answers is not in the document]
Comment (pulkitsinghall) [reply recorder/extensions/chrome/public/EventPage.js:132]: done

File: Daksha_Recorder_with_UI/public/ContentScript.js (hunk 7 above)
Comment (pulkitsinghall) [reply Daksha_Recorder_with_UI/public/ContentScript.js]: done

File: recorder/extensions/chrome/public/RecordingTab.js (hunk 15 above)
Comment (pulkitsinghall) [reply recorder/extensions/chrome/public/RecordingTab.js:2]: done

File: recorder/extensions/chrome/src/Components/dakshaRecorderEndPage.js (hunk 16 above)
Comment (pulkitsinghall) [reply recorder/extensions/chrome/src/Components/dakshaRecorderEndPage.js]: done

File: recorder/extensions/chrome/public/index.html [hunk 18]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Comment (prakashiit) [inline recorder/extensions/chrome/public/index.html:9]: Can we delete this file? If not, please update description

File: recorder/extensions/chrome/src/Components/dakshaRecorderCustomHardwaitPage.js [hunk 19]
```diff
@@ -0,0 +1,49 @@
+/*global chrome*/
+import React, { useState } from 'react';
+import '../ComponentCss/dakshaRecorderCustomHardwaitPage.css';
+let dakshaRecorderStartingPage = 1;
+let dakshaRecorderMainPage = 2 ;
```
Comment (AlphaRahul16) [inline recorder/extensions/chrome/src/Components/dakshaRecorderCustomHardwaitPage.js:5]: Can we make sometype of enum of these , to use in every component

File: recorder/extensions/chrome/src/App.js [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+  const [state, setState] = useState(pageNumber);
+
+  useEffect(() => {
+    var dakshaYamlObject = {};
+    dakshaYamlObject["popupPageNumber"] = state;
+    chrome.storage.sync.set(dakshaYamlObject, () => {
+    });
+  }, [state]);
+
+  const [image, changeImage] = useState(iconChanger);
```
Comment (AlphaRahul16) [inline recorder/extensions/chrome/src/App.js:37]: realign code

**Review by AlphaRahul16 — DISMISSED** [review]

**Review by hriddhighosh4 — APPROVED** [review]

**Review by prakashiit — DISMISSED** [review]

**Review by hriddhighosh4 — APPROVED** [review]

**Review by hriddhighosh4 — APPROVED** [review]
Approved

**Pull request merged**